### PR TITLE
Updated Rebellion class

### LINKED ISSUE #859: rebellion fallen angel animation
the problem is the rebellion jumps while casting, then he falls to early into the ground even though he's not done casting the skill, and when he's done he'll jump once again then fall onto the target.  its too obvious when ur rebellion has 1 dex (slow casting time)

correct me if im wrong, the animation should be like this

-cast then skill and jump , after done casting then falls onto the target

### LINKED ISSUE #1449: Rebellion skill Binding trap does 1 damage when using  Flicker
I think the title pretty much explains it. I'm using rAthena latest version. 

Steps to reproduce: Use binding trap, someone else steps on it, use Flicker and it does 1 HP damage. Also it slows down the caster (but i don't know if this is the correct behavior or not) and it seems like you can't cast on the same cell of someone else or yourself.

### PR DESCRIPTION
**Pull request by aleos89**
- Information based on Rebellion Awakening patch from kRO.
  - http://ro.gnjoy.com/news/update/View.asp?seq=183&curpage=1
- Fixes #859 and fixes #1449.
- There are a couple TODO but they are minor issues.

Thanks to @RagnarokNova and Fyrus for their information!

### CODE REVIEW

**Comment by zackdreaver** [comment]
Nice!
I've been working skill tree+description since its release, almost complete now.

**Comment by admkakaroto** [comment]
level base and class has changed?

Ref: http://ragnarokonline.gungho.jp/special/rebellion/about/

**Comment by zackdreaver** [comment]
@admkakaroto 
don't use jRO as source.

**Comment by zackdreaver** [comment]
I've added the translated skills and states
let me know if there something miss or wrong.
https://github.com/zackdreaver/ROenglishRE/commit/8c6a0c71485e78d8d3e5df82db1bb3088a667b14

**Comment by Felleonel** [comment]
Rebellion Base and Class Level is now 175/60 in kRO.  Kagerou and Oboro was updated too (Base/Class Level and HP/SP).

The reference is the same that @aleos89 published.

Link: http://ro.gnjoy.com/news/update/View.asp?seq=183&curpage=1
Print: http://prnt.sc/cnm3hy

I don't know korean, but I believe that's it.

**Comment by aleos89** [comment]
That is one of the TODO marks. We don't have the EXP table for it. It can be expanded but the EXP values won't be official.

**Comment by admkakaroto** [comment]
@aleos89 so to make the change base class and still need to change something basehp table? or just experience?

**Comment by admkakaroto** [comment]
@Lemongrass3110 

Whenever I use the 'RL Fallen_Angel' skill, received this message in map_server.

http://imgur.com/E1bGfN5

**Comment by admkakaroto** [comment]
@aleos89 l

Whenever I use the 'RL Fallen_Angel' skill, received this message in map_server.

http://imgur.com/E1bGfN5

The character goes to where it should, but does not activate the other!

**Comment by admkakaroto** [comment]
I reversed that part is It worked again, just do not know if it is correct official form.

https://github.com/rathena/rathena/pull/1552/files#diff-a62a8cdf5ca5dce3e8091f667d78f417L12165

https://github.com/rathena/rathena/pull/1552/files#diff-a62a8cdf5ca5dce3e8091f667d78f417L4142

**Comment by aleos89** [comment]
Thank you for the report @admkakaroto. I have fixed it in ecebd2c.

File: src/map/battle.c [hunk 1]
```diff
@@ -4368,6 +4401,13 @@ struct Damage battle_attack_sc_bonus(struct Damage wd, struct block_list *src, s
 			ATK_ADD(wd.equipAtk, wd.equipAtk2, sc->data[SC_DRUMBATTLE]->val2);
 		if (sc->data[SC_MADNESSCANCEL])
 			ATK_ADD(wd.equipAtk, wd.equipAtk2, 100);
+		if (sc->data[SC_MAGICALBULLET]) {
```
Comment (Darkelfen) [inline src/map/battle.c]: got compiling warning at this line:

battle.c: In function 'battle_attack_sc_bonus':
battle.c:4472: error: 'tstatus' undeclared (first use in this function)
battle.c:4472: error: (Each undeclared identifier is reported only once
battle.c:4472: error: for each function it appears in.)

Added: struct status_data *tstatus = status_get_status_data(target); and it compiled.

Comment (aleos89) [inline src/map/battle.c]: I'm curious why this is failing now when the initial commit compiled just fine and I haven't touched this skill in follow ups.

Comment (aleos89) [inline src/map/battle.c]: The PR is failing. The top two builds are always PR and the bottom two are Push.

Comment (cydh) [inline src/map/battle.c]: ok, when this branch merged to master, there are many errors :P

**Comment by cydh** [comment]
@aleos89 maybe just remake this branch, I tried rebase this on local, the log is crazy as hell.
if u see the file changes, it's unbelievable. xD just like this branch lost its track to master. (so does with my sc_config branch)

**Comment by cydh** [comment]
@aleos89 I wonder if u can give us full translation for 
> * Information based on Rebellion Awakening patch from kRO.
   * http://ro.gnjoy.com/news/update/View.asp?seq=183&curpage=1

File: db/pre-re/skill_db.txt [hunk 2]
```diff
@@ -1263,25 +1263,25 @@
 // Rebellion
 //2551,0,0,0,0,0,0,10,0,no,0,0,0,none,0,0,	RL_GLITTERING_GREED,Flip The Coin Greed
 2552,0,6,4,0,0x1,0,1,1,no,0,0,0,none,0,0x0,	RL_RICHS_COIN,Rich's Coin
-2553,15,6,1,-1,0,0,5,1,no,0,0,0,weapon,0,0x0,	RL_MASS_SPIRAL,Mass Spiral
-2554,9,6,1,-1,0x2,2,5,1,no,0,0,0,weapon,0,0x10020,	RL_BANISHING_BUSTER,Banishing Buster
-2555,3,6,2,0,0x1,1,5,1,no,0,0,3,misc,0,0x0,	RL_B_TRAP,Bind Trap
+2553,-9,6,1,-1,0,0,5,1,no,0,0,0,weapon,0,0x0,	RL_MASS_SPIRAL,Mass Spiral
+2554,-9,6,1,-1,0x1,0,5,1,no,0,0,0,weapon,0,0x10020,	RL_BANISHING_BUSTER,Banishing Buster
```
Comment (Darkelfen) [inline db/pre-re/skill_db.txt]: I think 0x1 here is wrong since this is a damaging skill.

Comment (aleos89) [inline db/pre-re/skill_db.txt]: Thanks, I've fixed this in the other PR.

**Comment by aleos89** [comment]
Moved this to #1692.

**Pull request closed without merging**